app/logical/sources/strategies/hentai_foundry.rb

```diff
@@ -0,0 +1,116 @@
+# Image URLs
+#
+# * https://pictures.hentai-foundry.com/a/Afrobull/795025/Afrobull-795025-kuroeda.png
+#
+# Page URLs
+#
+# * https://www.hentai-foundry.com/pictures/user/Afrobull/795025/kuroeda
+# * https://www.hentai-foundry.com/pictures/user/Afrobull/795025o
+#
+# Preview URLs
+#
+# * https://thumbs.hentai-foundry.com/thumb.php?pid=795025&size=350
+#
+# Profile URLs
+#
+# * https://www.hentai-foundry.com/user/kajinman/profile
+# * https://www.hentai-foundry.com/pictures/user/kajinman
+# * https://www.hentai-foundry.com/pictures/user/kajinman/scraps
+# * https://www.hentai-foundry.com/user/J-likes-to-draw/profile
+
+module Sources
+  module Strategies
+    class HentaiFoundry < Base
+      BASE_URL =    %r!\Ahttps?://(?:www\.)?hentai-foundry\.com!i
+      PAGE_URL =    %r!#{BASE_URL}/pictures/user/(?<artist_name>[\w-]+)/(?<illust_id>\d+)(?:/[\w-]*)?\z!i
+      PROFILE_URL = %r!#{BASE_URL}/(?:pictures/)?user/(?<artist_name>[\w-]+)(?:/[a-z]*)?\z!i
+      IMAGE_URL =   %r!\Ahttps?://pictures\.hentai-foundry\.com/\w/(?<artist_name>[\w-]+)/(?<illust_id>\d+)/[\w-]+\.\w+\z!i
```

Comment:
[This code](https://github.com/danbooru/danbooru/blob/86e3bd8fbc6b11cee543e7cee3747b907456c9af/app/models/post.rb#L373-L374) could theoretically be deduplicated now, somewhat like [this](https://github.com/danbooru/danbooru/blob/86e3bd8fbc6b11cee543e7cee3747b907456c9af/app/models/post.rb#L362-L363)

Reply:
Should never, under any circumstances, cause any web requests, though, if you do this.
That's why the method for DA is named <code>page_url_<b>from_image_url</b></code>

Reply:
Yeah, I didn't want to touch that part of the code because evazion said it all needs a revamp, and the normalization is already being done for this site there. Maybe in a future pull request.

Reply:
Leaving `Post#normalized_source` alone is fine for now, that can be cleaned up at some later point.